### PR TITLE
Correct vim modeline in ChangeLog

### PR DESCRIPTION
**Pull request by softins**
**Short description of changes**

Update the vim modeline in `ChangeLog` to fix vim error and enable syntax highlighting

CHANGELOG: SKIP

**Context: Fixes an issue?**

The vim modeline that was added to `ChangeLog` in #2899 is enough to make Github render the file as Markdown, but it results in vim errors when editing the file:

```
$ vim ChangeLog
"ChangeLog" 2056L, 73367B
Error detected while processing modelines:
line 2056:
E518: Unknown option: -->
Press ENTER or type command to continue
```

It is because the closing comment tag `-->` is interpreted by vim as a command. In addition, the `syntax=Markdown` does not turn on syntax highlighting within vim. For that, `set ft=markdown:` is needed, and the trailing `:` prevents the `-->` from being interpreted by vim.

**Does this change need documentation? What needs to be documented and how?**

**Status of this Pull Request**

Ready to merge

**What is missing until this pull request can be merged?**

Nothing

## Checklist

-  [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
-  [x] I tested my code and it does what I want
-  [x] My code follows the [style guide](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#source-code-consistency) 
-  [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
-  [x] I've filled all the content above

### CODE REVIEW

**Review by ann0see — APPROVED** [review]
Can confirm that it works.

File: ChangeLog [hunk 1]
```diff
@@ -2053,4 +2053,4 @@
 
 - initial version
 
-<!-- vim: syntax=Markdown -->
+<!-- vim: syntax=Markdown: set ft=markdown: -->
```
Comment (pljones) [inline ChangeLog:2056]: Maybe use the full form of the setting - `filetype` - rather than an abbreviation?

Comment (pljones) [reply ChangeLog:2056]: (Could do with `filetype=cfg` in `Jamulus.pro`, too.)

Comment (softins) [reply ChangeLog:2056]: > (Could do with `filetype=cfg` in `Jamulus.pro`, too.)

Good idea. Though I'll do that separately from this bug fix.

Comment (softins) [reply ChangeLog]: > (Could do with `filetype=cfg` in `Jamulus.pro`, too.)

`cfg` is an ok generic filetype for `Jamulus.pro`, but better is to install the syntax and filetype detect files from https://github.com/artoj/qmake-syntax-vim

**Review by pljones — APPROVED** [review]

**Pull request merged**